geode-core/src/main/java/org/apache/geode/internal/cache/tier/InterestType.java

```diff
@@ -14,36 +14,36 @@
  */
 package org.apache.geode.internal.cache.tier;
 
+import org.jetbrains.annotations.NotNull;
+
 /**
- * Pre-defined interest list types supported by the system
- *
- * @version $Revision: 13166 $
- * @since GemFire 2.0.2
+ * Pre-defined interest list types supported by the system.
+ * Partial duplicate of {@link org.apache.geode.cache.operations.InterestType}.
  */
-public class InterestType {
+public enum InterestType {
 
```

Comment:
Maybe use Key(0), REGULAR_EXPRESSION(1) ...; int value; InterestType(int value) { this.value=value; } is a little bit better?

Just comments.